### PR TITLE
Implement create interface for filedb database

### PR DESCRIPTION
**Pull request by khanhtc1202**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/datastore/applicationstore.go [hunk 1]
```diff
@@ -54,6 +55,52 @@ func (a *applicationCollection) GetUpdatableShard() (Shard, error) {
 	}
 }
```
Comment (nghialv) [inline pkg/datastore/applicationstore.go:57]: Please add a TODO to generate those kinds of code automatically by specifying some field in the proto file.

https://github.com/pipe-cd/pipecd/blob/master/pkg/model/application.proto#L24

For example,

``` proto
message Application {
    // The generated unique identifier.
    string id = 1 [(validate.rules).string.min_len = 1, shard=client];

```

**Comment by nghialv** [comment]
Good job.
/lgtm

File: pkg/datastore/applicationstore.go (hunk 1 above)
Comment (khanhtc1202) [reply pkg/datastore/applicationstore.go:57]: Nice idea 👍

File: pkg/datastore/eventstore.go [hunk 2]
```diff
@@ -50,6 +52,23 @@ func (e *eventCollection) GetUpdatableShard() (Shard, error) {
 	}
 }
 
+func (e *eventCollection) Encode(entity interface{}) (map[Shard][]byte, error) {
+	errFmt := "failed while encode Event object: %s"
```
Comment (knanao) [inline pkg/datastore/eventstore.go:56]:
```suggestion
	const errFmt = "failed while encode Event object: %s"
```

File: pkg/datastore/pipedstore.go [hunk 3]
```diff
@@ -50,6 +52,23 @@ func (p *pipedCollection) GetUpdatableShard() (Shard, error) {
 	}
 }
 
+func (p *pipedCollection) Encode(e interface{}) (map[Shard][]byte, error) {
+	errFmt := "failed while encode Piped object: %s"
```
Comment (knanao) [inline pkg/datastore/pipedstore.go:56]:
```suggestion
	const errFmt = "failed while encode Piped object: %s"
```

File: pkg/datastore/projectstore.go [hunk 4]
```diff
@@ -50,6 +52,23 @@ func (p *projectCollection) GetUpdatableShard() (Shard, error) {
 	}
 }
 
+func (p *projectCollection) Encode(e interface{}) (map[Shard][]byte, error) {
+	errFmt := "failed while encode Project object: %s"
```
Comment (knanao) [inline pkg/datastore/projectstore.go:56]:
```suggestion
	const errFmt = "failed while encode Project object: %s"
```

File: pkg/datastore/commandstore.go [hunk 5]
```diff
@@ -86,6 +86,35 @@ func (c *commandCollection) Decode(e interface{}, parts ...[]byte) error {
...
+func (c *commandCollection) Encode(e interface{}) (map[Shard][]byte, error) {
+	errFmt := "failed while encode Command object: %s"
+
+	me, ok := e.(*model.Command)
+	if !ok {
+		return nil, fmt.Errorf(errFmt, "type not matched")
+	}
+
+	agentShardStruct := me
+	adata, err := json.Marshal(&agentShardStruct)
```
Comment (knanao) [inline pkg/datastore/commandstore.go:98]: Is it to improve more readability? if so, it's okay.
```suggestion
	adata, err := json.Marshal(me)
```

Comment (khanhtc1202) [reply pkg/datastore/commandstore.go:98]: yes, it's 👍 lets me keep it as is to make it more clear that we're building data for agent shard 😄

Comment (knanao) [reply pkg/datastore/commandstore.go:98]: I got you. Pretty good!

**Comment by knanao** [comment]
Nice catch!
/approve

**Pull request merged**